### PR TITLE
chore: add `eslint-plugin-react`

### PR DESCRIPTION
**Pull request by marcalexiei**
## Context & Description

- add `eslint-plugin-react`
- change `packages/*` `tsconfig.json#jsx` from `react` to the modern `react-jsx`
- refactored `packages/router/src/components/Link.tsx` to be more `react/prop-types` friendly

### CODE REVIEW

File: packages/router/src/components/Link.tsx [hunk 1]
```diff
@@ -1,49 +1,61 @@
...
-import type { AnchorHTMLAttributes } from 'react'
+import type * as React from 'react'
 import { useInView } from 'react-intersection-observer'
 
 import { useRouter } from '../hooks/useRouter'
 import useRoute from '../hooks/useRoute'
 
-interface TuonoLinkProps {
+interface TuonoLinkProps
+  extends Omit<React.AnchorHTMLAttributes<HTMLAnchorElement>, 'preload'> {
```
Comment (Valerioageno) [inline packages/router/src/components/Link.tsx:8]: Does `React.AnchorHTMLAttributes<HTMLAnchorElement>` have the `preload` property?

File: packages/router/src/components/Link.tsx [hunk 2]
```diff
@@ -1,49 +1,61 @@
...
+  componentProps: TuonoLinkProps,
 ): React.JSX.Element {
-  const { preload = true, scroll = true, ...props } = componentProps
+  const {
+    preload = true,
+    scroll = true,
+    children,
+    href,
+    onClick,
+    ...otherProps
```
Comment (Valerioageno) [inline packages/router/src/components/Link.tsx:30]: I think `...rest` is more React idiomatic

**Comment by Valerioageno** [comment]
Just left few comments. LGTM

File: packages/router/src/components/Link.tsx (hunk 1 above)
Comment (marcalexiei) [reply packages/router/src/components/Link.tsx:8]: No, I don't know how got in my mind 😩

File: packages/router/src/components/Link.tsx (hunk 2 above)
Comment (marcalexiei) [reply packages/router/src/components/Link.tsx:30]: Yo are right

**Review by Valerioageno — APPROVED** [review]
LGTM!

**Pull request merged**